src/main/java/com/sarality/list/impl/IconListItemRenderer.java

```diff
@@ -0,0 +1,67 @@
+package com.sarality.list.impl;
+
+import android.view.View;
+import android.widget.CheckBox;
+
+import com.sarality.list.R;
+
+/**
+ * Implementation of the List View Item Renderer where each item has a secondary action
+ *
+ * @author satya (satya puniani)
+ */
+
+public abstract class IconListItemRenderer<T> extends CommonListItemRenderer<T, IconListItemViewHolder> {
+
+  public IconListItemRenderer(boolean displayLine2, boolean displayLine3) {
+    this(R.layout.icon_list_item, displayLine2, displayLine3);
+  }
+
+  public IconListItemRenderer(int itemLayoutId, boolean displayLine2, boolean displayLine3) {
+    super(itemLayoutId, displayLine2, displayLine3);
+  }
+
+  @Override
+  public void render(final View view, final IconListItemViewHolder viewHolder, final int position, final T data) {
+    super.render(view, viewHolder, position, data);
+
+    viewHolder.iconActionCheckBox.setChecked(getIconActionState(position, data));
+
+    int drawableResId = getIconDrawable(position, data);
+    if (drawableResId > 0) {
+      viewHolder.iconActionCheckBox.setButtonDrawable(drawableResId);
+    }
+
+    View.OnClickListener secondaryActionListener = new View.OnClickListener() {
```

Comment:
As discussed on Friday, splitting the checkbox action from the UI refresh. This is the minimal solution, which checks/unchecks the checkbox, and exposes an abstract method that the implementor can use to either perform the action, or block the action.

Reply:
updated the branch with the changes to the View Holder and added a listener for the checkbox on each row.